src/Definition/Reflection/Parameter.php

```diff
@@ -63,11 +63,14 @@ public function getPosition() : int
      */
     public function getType() : ?string
     {
-        if ($this->reflection->hasType()) {
-            return $this->reflection->getType()->getName();
+        if (! $this->reflection->hasType()) {
+            return null;
         }
 
-        return null;
+        $type = $this->reflection->getType();
+        assert($type !== null);
```

Comment:
PHPstan does not infer that the returned type is not null when `hasType()` is `true`. I guess asserting not-null is safe here.

Reply:
In my opinion this should be added then to the exceptions in phpstan configuration.

Reply:
I'll try a more recent version of phpstan first. Adding ignores is also something I'd like to avoid when possible.

Maybe switching to psalm would also be an option. It feels a bit more reliable in comparison. At least it's type assertions and inference is quite powerful.

Reply:
Upgrading PHPStan does not fix it and it will result in a bigger PR.

I reverted the code change and added an exclude to phpstan.neon, but this is awful. I'd like to rewrite this method like this:

```php
    public function getType() : ?string
    {
        $type = $this->reflection->getType();
        return $type !== null ? $type->getName() : null;
    }
```

What do you think?